libkbfs/block_journal.go

```diff
@@ -193,6 +193,62 @@ func makeBlockJournal(
 	return journal, nil
 }
 
+// The functions below are for reading and writing aggregate info.
+
+// Ideally, this would be a JSON file, but we'd need a JSON
+// encoder/decoder that supports unknown fields.
+type aggregateInfo struct {
+	UnflushedBytes int64
+
+	codec.UnknownFieldSetHandler
+}
+
+func (j *blockJournal) aggregateInfoPath() string {
+	return filepath.Join(j.dir, "aggregate_info")
+}
+
+func (j *blockJournal) getAggregateInfo() (aggregateInfo, error) {
+	data, err := ioutil.ReadFile(j.aggregateInfoPath())
+	if os.IsNotExist(err) {
+		return aggregateInfo{}, nil
+	} else if err != nil {
+		return aggregateInfo{}, err
+	}
+
+	var info aggregateInfo
+	err = j.codec.Decode(data, &info)
+	if err != nil {
+		return aggregateInfo{}, err
+	}
+
+	return info, nil
```

Comment:
Seems like it would be useful and straightforward to cache this in `j` so we don't read it from disk every time.